lib/src/message.dart

```diff
@@ -13,11 +14,39 @@ class Message {
     required this.ref,
   });
 
-  Map<String, dynamic> toJson() => {
-        'topic': topic,
-        'event':
-            event != ChannelEvents.heartbeat ? event.eventName() : 'heartbeat',
-        'payload': payload,
-        'ref': ref
-      };
+  /// Converting to JSON while removing functions
+  Map<String, dynamic> toJson() {
+    late final dynamic processedPayload;
+    if (payload is Map) {
+      processedPayload = <String, dynamic>{};
+      for (final outerKey in payload.keys) {
+        final outerValue = payload[outerKey];
+        if (outerValue is Map) {
+          for (final innerKey in outerValue.keys) {
+            final innerValue = outerValue[innerKey];
+            processedPayload[outerKey] ??= {};
+            if (innerValue is Binding) {
+              processedPayload[outerKey][innerKey] = <String, dynamic>{
+                'type': innerValue.type,
+                'filter': innerValue.filter,
+              };
+            } else {
+              processedPayload[outerKey][innerKey] = innerValue;
+            }
+          }
+        } else {
+          processedPayload[outerKey] = outerValue;
+        }
+      }
```

Comment:
The reason why this `toJson` method is such a mess is because `payload` might contain functions, which throws an error when trying to convert it using `JSON.encode`. This `toJson` essentially is removing all the functions contained in the `payload`.